src/executables/util_cs.c

```diff
@@ -4649,9 +4652,71 @@ memmon (UTIL_FUNCTION_ARG * arg)
       goto error_exit;
     }
 
+  /* execute phase */
+  error_code = mmon_get_server_info (server_info);
+  if (error_code != NO_ERROR)
+    {
+      goto error_exit;
+    }
+
+  // TODO: mmon_print_server_info (server_info);
+
+  if (module_index >= 0)
+    {
+      error_code = mmon_get_module_info (module_index, module_info);
+      if (error_code != NO_ERROR)
+	{
+	  goto error_exit;
+	}
+
+      // TODO: mmon_print_module_info (module_info);
+    }
+  else
+    {
+      if (print_show_all)
+	{
+	  module_count = MMON_MODULE_LAST;
+	  print_transaction = true;
+	}
+      else if (!print_transaction && !print_module)
```

Comment:
Nit:

The print_transaction variable is set in the if block. However, it is used as a condition of the else if. It looks a bit awkward. I recommand you to set in advance all related condition variables like print_transaction and information variables like tran_count.

Additionally, how about introducing a flag variable like is_module_summary?
```
if (is_module_summary)
{
    mmon_get_module_info_summary()
}
else
{
    mmon_get_module_info()
}
```

Reply:
I add the flag `print_default` and set the flag at the end of input check phase.